components/select_team/index.js

```diff
@@ -9,17 +9,37 @@ import {getTeams} from 'mattermost-redux/actions/teams';
 import {loadRolesIfNeeded} from 'mattermost-redux/actions/roles';
 import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
 import {getRoles} from 'mattermost-redux/selectors/entities/roles';
+import {getJoinableTeams, getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 
 import SelectTeam from './select_team.jsx';
 
 function mapStateToProps(state) {
     const license = getLicense(state);
     const config = getConfig(state);
+    const currentUser = getCurrentUser(state);
+    const myTeamMemberships = Object.values(getTeamMemberships(state));
+
+    function sortTeams(a, b) {
+        const options = {
+            numeric: true,
+            sensitivity: 'base',
+        };
+        return a.display_name.localeCompare(b.display_name, currentUser.locale || 'en', options);
+    }
+
+    const joinableTeams = Object.values(getJoinableTeams(state)).
+        filter((team) => team.delete_at === 0).
+        sort(sortTeams);
```

Comment:
This is going to be a new object every time, probably better to make a selector to do this

Reply:
Thanks!  I added selector for that.